frontend/src/components/home-page/question-types/multiple-choice.tsx

```diff
@@ -1,40 +1,99 @@
-import React, { useState, useEffect } from "react";
-import './multiple-choice.scss'
+import React, { useState, useEffect, useContext } from 'react';
+import { store } from '../../../store';
+import './multiple-choice.scss';
 
 type Props = {
 	answers: string[];
-}
+	correct: number;
+	responses: string[];
+};
 
 type AnswerChoiceProps = {
 	answer: string;
 	letter: string;
-}
+	correct: boolean;
+	response: string;
+	index: number;
+};
+
+const AnswerChoice = ({
+	answer,
+	letter,
+	correct,
+	response,
+	index,
+}: AnswerChoiceProps) => {
+	const global = useContext(store) as any;
+	const dispatch = global.dispatch;
+	const state = global.state;
+
+	const editQuestion = (event: any, index: number) => {
+		const newQuestions = state.questions.slice();
+		newQuestions[state.preview[0]].questions[state.preview[1]].choices[index] =
```

Comment:
For state.preview in this file, the naming needs to be more descriptive. It isn't clear what state.preview[0] and state.preview[1] are. These should be broken up into two separate things with descriptive names. It also doesn't seem to me like state.preview is being changed in this component, so it could be passed as props, with state.preview being managed in a parent component (although if it is global state I guess that doesn't really matter.).